apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs

```diff
@@ -608,43 +622,50 @@ public async Task DMLReturn_ExecuteNonQueryAsync()
             Assert.Equal(expected, actual);
         }
 
-        [Fact(Skip = "This test is skipped until DML Returning changes are done.")]
+        // Without DML's return clause also ExecuteReaderAsync should update the database table.
+        // First ReadAsync call should return false in the absence of 'return` clause.
+        [Fact]
         public async Task NoDMLReturn_ExecuteReader_Read()
         {
             string key = _fixture.CreateTestRows();
             var actualValues = new List<int>();
 
             using var connection = _fixture.GetConnection();
             await connection.OpenAsync();
-            RetryHelpers.ExecuteWithRetry(async () =>
+            bool readOnce = false;
+            string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
+            var command = connection.CreateDmlCommand(dml);
+            command.Parameters.Add("key", SpannerDbType.String, key);
+            using var reader = await command.ExecuteReaderAsync();
+            readOnce = await reader.ReadAsync();
+
+            var actual = _fixture.FetchValues(key);
+            var expected = new Dictionary<int, int>
             {
-                string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
-                var command = connection.CreateDmlCommand(dml);
-                using var reader = await command.ExecuteReaderAsync();
-                while (await reader.ReadAsync())
-                {
-                    // TODO: Yet to decide behaviour.
-                    //actualValues.Add(reader.GetFieldValue<int>(reader.GetOrdinal("Value")));
-                }
-            });
-            var expectedValues = new List<int> { 2, 5 };
-            Assert.Equal(expectedValues, actualValues);
+                { 0, 0 }, // Not updated
+                { 1, 2 }, // Updated
+                { 2, 2 }, // Not updated
+                { 3, 3 }, // Not updated
+                { 4, 5 }  // Updated
+            };
+
+            Assert.False(readOnce);
+            Assert.Equal(expected, actual);
         }
 
-        [Fact(Skip = "This test is skipped until DML Returning changes are done.")]
+        // Without DML's return clause and ReadAsync call also ExecuteReaderAsync should update database table.
+        [Fact]
```

Comment:
We do still need another test for what happens if we call ExecuteReader and *do* read from the returned reader without any "With Return" part.

Reply:
Yes, I wasn't suggesting removing this test - just adding another one.

Reply:
Great thanks.